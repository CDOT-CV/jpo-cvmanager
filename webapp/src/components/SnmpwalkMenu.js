import React, { useEffect } from 'react'
import SnmpwalkItem from './SnmpwalkItem'
import { useSelector, useDispatch } from 'react-redux'
import { confirmAlert } from 'react-confirm-alert'
import { Options } from './AdminDeletionOptions'
import { selectRsuManufacturer, selectRsuIpv4 } from '../slices/rsuSlice'
import {
    selectMsgFwdConfig,
    selectErrorState,

    // Actions
    refreshSnmpFwdConfig,
} from '../slices/configSlice'
import Button from '@mui/material/Button'
import DeleteIcon from '@mui/icons-material/Delete'
import './css/SnmpwalkMenu.css'
import {
    // Actions
    deleteSnmpSet,
} from '../slices/configSlice'

const SnmpwalkMenu = (props) => {
    const dispatch = useDispatch()

    const msgFwdConfig = useSelector(selectMsgFwdConfig)
    const errorState = useSelector(selectErrorState)

    const rsuIp = useSelector(selectRsuIpv4)
    const rsuManufacturer = useSelector(selectRsuManufacturer)

    useEffect(() => {
        // Refresh Data
        dispatch(refreshSnmpFwdConfig([rsuIp]))
    }, [rsuIp, dispatch])

    const handleDelete = (msgType, ip) => {
        const buttons = [
            {
                label: 'Yes',
                onClick: () => deleteSnmpSet(ip, msgType),
            },
            {
                label: 'No',
                onClick: () => {},
            },
        ]
        const alertOptions = Options(
            'Delete SNMP Configuration',
            `Are you sure you want to delete SNMP forwarding for ${msgType} messages to the IP: ${ip}? `,
            buttons
        )

        confirmAlert(alertOptions)
    }

    return (
        <div id="snmpdiv">
            <h2 id="snmpheader">Current Message Forwarding</h2>

            {rsuManufacturer === 'Yunex' ? (
                <div>
                    {Object.hasOwn(msgFwdConfig, 'rsuXmitMsgFwdingTable') &&
                    Object.hasOwn(msgFwdConfig, 'rsuReceivedMsgTable') ? (
                        <div>
                            <h2 id="snmptxrxheader">TX Forward Table</h2>
                            {Object.keys(
                                msgFwdConfig.rsuXmitMsgFwdingTable
                            ).map((index) => (
                                <div key={'msgFwd-' + index}>
                                    <Button
                                        className="deletebutton"
                                        onClick={() =>
                                            handleDelete(
                                                msgFwdConfig
                                                    .rsuXmitMsgFwdingTable[
                                                    index
                                                ]['Message Type'],
                                                msgFwdConfig
                                                    .rsuXmitMsgFwdingTable[
                                                    index
                                                ]['IP'],
                                                index
                                            )
                                        }
                                        startIcon={<DeleteIcon />}
                                    >
                                        Delete
                                    </Button>
                                    <SnmpwalkItem
                                        key={'snmptxitem-' + index}
                                        content={
                                            msgFwdConfig.rsuXmitMsgFwdingTable[
                                                index
                                            ]
                                        }
                                        index={index}
                                    />
                                </div>
                            ))}
                            <h2 id="snmptxrxheader">RX Forward Table</h2>
                            {Object.keys(msgFwdConfig.rsuReceivedMsgTable).map(
                                (index) => (
                                    <div>
                                        <Button
                                            className="deletbutton"
                                            onClick={() =>
                                                handleDelete(
                                                    msgFwdConfig
                                                        .rsuReceivedMsgTable[
                                                        index
                                                    ]['Message Type'],
                                                    msgFwdConfig
                                                        .rsuReceivedMsgTable[
                                                        index
                                                    ]['IP'],
                                                    index
                                                )
                                            }
                                            startIcon={<DeleteIcon />}
                                        >
                                            Delete
                                        </Button>
                                        <SnmpwalkItem
                                            key={'snmprxitem-' + index}
                                            content={
                                                msgFwdConfig
                                                    .rsuReceivedMsgTable[index]
                                            }
                                            index={index}
                                        />
                                    </div>
                                )
                            )}
                        </div>
                    ) : null}
                </div>
            ) : (
                <div>
                    {Object.keys(msgFwdConfig).map((index) => (
                        <div>
                            <Button
                                className="deletbutton"
                                onClick={() =>
                                    handleDelete(
                                        msgFwdConfig[index]['Message Type'],
                                        msgFwdConfig[index]['IP'],
                                        index
                                    )
                                }
                                startIcon={<DeleteIcon />}
                            >
                                Delete
                            </Button>
                            <SnmpwalkItem
                                key={'snmpitem-' + index}
                                content={msgFwdConfig[index]}
                                index={index}
                            />
                        </div>
                    ))}
                </div>
            )}

            {errorState !== '' ? <p id="warningtext">{errorState}</p> : <div />}
        </div>
    )
}

export default SnmpwalkMenu
