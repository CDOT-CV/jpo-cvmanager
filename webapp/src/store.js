import { configureStore } from "@reduxjs/toolkit";
import rsuReducer from "./slices/rsuSlice";
import userReducer from "./slices/userSlice";
import wzdxReducer from "./slices/wzdxSlice";
import configReducer from "./slices/configSlice";

const store = configureStore({
  reducer: {
    rsu: rsuReducer,
    user: userReducer,
    wzdx: wzdxReducer,
    config: configReducer,
  },
});

export default store;
