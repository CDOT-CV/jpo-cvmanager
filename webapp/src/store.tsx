import { Action, ThunkAction, configureStore } from '@reduxjs/toolkit'
import rsuReducer from './generalSlices/rsuSlice'
import userReducer from './generalSlices/userSlice'
import wzdxReducer from './generalSlices/wzdxSlice'
import configReducer from './generalSlices/configSlice'
import adminAddOrganizationReducer from './features/adminAddOrganization/adminAddOrganizationSlice'
import adminAddRsuReducer from './features/adminAddRsu/adminAddRsuSlice'
import adminAddUserReducer from './features/adminAdduser/adminAddUserSlice'
import adminEditOrganizationReducer from './features/adminEditOrganization/adminEditOrganizationSlice'
import adminEditRsuReducer from './features/adminEditRsu/adminEditRsuSlice'
import adminEditUserReducer from './features/adminEditUser/adminEditUserSlice'
import adminOrganizationTabReducer from './features/adminOrganizationTab/adminOrganizationTabSlice'
import adminOrganizationTabUserReducer from './features/adminOrganizationTabUser/adminOrganizationTabUserSlice'
import adminOrganizationTabRsuReducer from './features/adminOrganizationTabRsu/adminOrganizationTabRsuSlice'
import adminRsuTabReducer from './features/adminRsuTab/adminRsuTabSlice'
import adminUserTabReducer from './features/adminUserTab/adminUserTabSlice'
import menuReducer from './features/menu/menuSlice'
import intersectionMapReducer from './components/map/map-slice'
import intersectionMapLayerStyleReducer from './components/map/map-layer-style-slice'

export const setupStore = (preloadedState: any) => {
  return configureStore({
    reducer: {
      rsu: rsuReducer,
      user: userReducer,
      wzdx: wzdxReducer,
      config: configReducer,
      adminAddOrganization: adminAddOrganizationReducer,
      adminAddRsu: adminAddRsuReducer,
      adminAddUser: adminAddUserReducer,
      adminEditOrganization: adminEditOrganizationReducer,
      adminEditRsu: adminEditRsuReducer,
      adminEditUser: adminEditUserReducer,
      adminOrganizationTab: adminOrganizationTabReducer,
      adminOrganizationTabUser: adminOrganizationTabUserReducer,
      adminOrganizationTabRsu: adminOrganizationTabRsuReducer,
      adminRsuTab: adminRsuTabReducer,
      adminUserTab: adminUserTabReducer,
      menu: menuReducer,
      intersectionMap: intersectionMapReducer,
      intersectionMapLayerStyle: intersectionMapLayerStyleReducer,
    },
    preloadedState,
  })
}

type AppStore = ReturnType<typeof setupStore>
export type AppState = ReturnType<AppStore['getState']>
type AppThunk<ReturnType = void> = ThunkAction<ReturnType, AppState, unknown, Action>

export type RootState = ReturnType<ReturnType<typeof setupStore>['getState']>
