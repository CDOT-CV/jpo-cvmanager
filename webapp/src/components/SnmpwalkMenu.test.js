import React from "react";
import { render } from "@testing-library/react";
import SnmpwalkMenu from "./SnmpwalkMenu";
import { Provider } from "react-redux";
import { setupStore } from "../store";

it("should take a snapshot", () => {
  const { asFragment } = render(
    <Provider store={setupStore({})}>
      <SnmpwalkMenu />
    </Provider>
  );

  expect(
    asFragment(
      <Provider store={setupStore({})}>
        <SnmpwalkMenu />
      </Provider>
    )
  ).toMatchSnapshot();
});
