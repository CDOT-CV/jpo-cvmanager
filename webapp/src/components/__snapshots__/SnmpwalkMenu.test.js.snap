// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    id="snmpdiv"
  >
    <h2
      id="snmpheader"
    >
      Current Message Forwarding
    </h2>
    <div />
    <div />
  </div>
</DocumentFragment>
`;
